### DISCUSSION TITLE
Iterate markergroups

### DISCUSSION THREAD

**Discussion by hupe13** [bozdoz/wp-plugin-leaflet-map#133] [Show and tell]
Hi Bozdoz,

it is not a issue, it is a notice.
You describe in your FAQ:

> // iterate any of these: `maps`, `markers`, `markergroups`, `lines`, `circles`, `geojsons`
>     var maps = window.WPLeafletMapPlugin.maps;
>     for (var i = 0, len = maps.length; i < len; i++) {

That suggests you could do it that way:
```
var markergroups = window.WPLeafletMapPlugin.markergroups;
for (var i = 0, len = markergroups.length; i < len; i++) {
```
But the working code is:
```
var markergroups = window.WPLeafletMapPlugin.markergroups;
Object.entries(markergroups).forEach(([key, value]) => {
```

See example:
https://github.com/hupe13/extensions-leaflet-map/blob/6d13c3fb8c1c4415d1781d711c9003a3b301da95/php/zoomhome.php#L63

**Comment by bozdoz** [comment]
True.  Do you think it's worth updating?  Maybe I could update the jsdoc so that IDE's can see the types?

**Comment by hupe13** [comment]
I found this in connection with marker clusters and groups, markergroups.length was always 0. That's why I investigated that. 
I think, you should put that in the faq. 

> Maybe I could update the jsdoc so that IDE's can see the types?

Maybe.

**Reply by bozdoz** [reply]
Updated the README

**Comment by hupe13** [comment]
Thank you very much.
